flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecTemporalJoin.java

```diff
@@ -276,12 +278,21 @@ private TwoInputStreamOperator<RowData, RowData, RowData> createJoinOperator(
                     isLeftOuterJoin);
         } else {
             if (isTemporalFunctionJoin) {
-                return new TemporalProcessTimeJoinOperator(
-                        InternalTypeInfo.of(rightInputType),
-                        generatedJoinCondition,
-                        minRetentionTime,
-                        maxRetentionTime,
-                        isLeftOuterJoin);
+                if (config.get(ExecutionConfigOptions.TABLE_EXEC_ASYNC_STATE_ENABLED)) {
+                    return new AsyncStateTemporalProcessTimeJoinOperator(
```

Comment:
The Jira mentions SQL and table and the component in the PR title mentions [TABLE], but the fix appears to be datastream only. Is there an intent that this effects table API and SQL?  

Please could you update the documentation -as this is new changes to the API.

Reply:
Thank you for reviewing. This PR is for completing a part of Flip-473. His intention is similar to https://github.com/apache/flink/pull/25320 .

The original API has not been changed, it will only switch the TemporalProcessTimeJoinOperator under the FLINK-TABLE module to be executed by Asynchronous StateTemporalProcessTimeJoinOperator when table.exe.async-state.enable is set to true.

Is there anything else that needs to be modified? If so, please let me know. Thank you

Reply:
> Thanks for review. This PR is for completing a part of Flip-473. Its intention is similar to #25320 .
> 
> The original API has not been changed, it will only switch the TemporalProcessTimeJoinOperator under the FLINK-TABLE module to be executed by Asynchronous StateTemporalProcessTimeJoinOperator when _table. exec.async-state.enable_ is set to true.
> 
> Is there anything else that needs to be modified? If so, please let me know. Thank you

ok thanks for the clarification